bakery/src/tasks/assemble-book-group.js

```diff
@@ -0,0 +1,62 @@
+const dedent = require('dedent')
+
+const { constructImageSource } = require('../task-util/task-util')
+
+const task = (taskArgs) => {
+  const imageDefault = {
+    name: 'openstax/nebuchadnezzar',
+    tag: 'trunk'
+  }
+  const imageOverrides = taskArgs != null && taskArgs.image != null ? taskArgs.image : {}
+  const imageSource = constructImageSource({ ...imageDefault, ...imageOverrides })
+  const inputName = 'fetched-book-group'
+  const assembledOutput = 'assembled-book-group'
+  const symlinkOutput = 'module-symlinks'
+  const rawCollectionDir = `${inputName}/raw`
+
+  return {
+    task: 'assemble book',
+    config: {
+      platform: 'linux',
+      image_resource: {
+        type: 'docker-image',
+        source: imageSource
+      },
+      inputs: [
+        { name: inputName }
+      ],
+      outputs: [
+        { name: assembledOutput },
+        { name: symlinkOutput }
+      ],
+      run: {
+        path: '/bin/bash',
+        args: [
+          '-cxe',
+          dedent`
+          exec 2> >(tee ${assembledOutput}/stderr >&2)
+          for collection in $(find "${rawCollectionDir}/collections/" -type f); do
+            slug_name=$(basename "$collection" | awk -F'[.]' '{ print $1; }')
+            rm -rf temp-assembly
+
+            mv "$collection" "${rawCollectionDir}/modules/collection.xml"
+            mv "${rawCollectionDir}/metadata/$slug_name.metadata.json" "${rawCollectionDir}/modules/metadata.json"
+            
+            # Assembly destination must nested EXACTLY one level under cwd for symlinks to work 
+            neb assemble "${rawCollectionDir}/modules" temp-assembly/
+
+            # We shouldn't we need this symlink
+            rm temp-assembly/collection.xml
+
+            find temp-assembly -type l | xargs -I{} cp -P {} "${symlinkOutput}"
+            find "${symlinkOutput}" -type l | xargs -I{} cp -P {} "${assembledOutput}"
```

Comment:
Question / note for myself: The why of this will probably become more obvious as I get to where they're used, but it seems we're populating the same symlinks in both outputs.

Reply:
To potentially help with understanding, there are two reasons to add symlinks:
1. A step needs to resolve links to resources. This is for correctness.
2. The symlinks will allow images to show if a developer opens a markup file containing links between steps. This is for developer convenience, not correctness.

If we dump symlinks to an input to a task, it's probably reason 1 (and you'll notice `fetched-book` is also included as an input - so the links have something to resolve to). If we dump symlinks into an output, it's probably reason 2 (because the links will actually make their way into the output directory in the file system, and it's is not necessary to include `fetched-book` as an input to the task)

Reply:
Cool, thanks for the breakdown on that! I suspect the symlinks on this on are reason 2 as I didn't see them being used anywhere later in scripts, but it makes sense that it would be convenient to see images when you view the XHTML.